functions/Invoke-DbaDiagnosticQuery.ps1

```diff
@@ -79,235 +88,341 @@ Run the selection made by the user on the Sql Server instance specified.
+		[switch][Alias('Silent')]
+		$EnableException
+		, [string]$OutputPath
+		, [switch]$ExportQueries
+	)
+
+	begin {
+		[int]$ProgressId = Get-Random
+		[System.Collections.Generic.List[pscustomobject]]$QueryDetail = @()
+		function Invoke-DiagnosticQuerySelectionHelper {
+			[CmdletBinding()]
+			Param (
+				[parameter(Mandatory = $true)]
+				$ParsedScript
+			)
+
+			$ParsedScript | Select-Object QueryNr, QueryName, DBSpecific, Description | Out-GridView -Title "Diagnostic Query Overview" -OutputMode Multiple | Sort-Object QueryNr | Select-Object -ExpandProperty QueryName
+
+		}
+
+		Write-Message -Level Verbose -Message "Interpreting DMV Script Collections"
+
+		$module = Get-Module -Name dbatools
+		$base = $module.ModuleBase
+
+		if (!$Path) {
+			$Path = "$base\bin\diagnosticquery"
+		}
+
+		$scriptversions = @()
+		$scriptfiles = Get-ChildItem "$Path\SQLServerDiagnosticQueries_*_*.sql"
+
+		if (!$scriptfiles) {
+			Write-Message -Level Warning -Message "Diagnostic scripts not found in $Path. Using the ones within the module."
+
+			$Path = "$base\bin\diagnosticquery"
+
+			$scriptfiles = Get-ChildItem "$base\bin\diagnosticquery\SQLServerDiagnosticQueries_*_*.sql"
+			if (!$scriptfiles) {
+				Stop-Function -Message "Unable to download scripts, do you have an internet connection? $_" -ErrorRecord $_
+				return
+			}
+		}
+
+		[int[]]$filesort = $null
+
+		foreach ($file in $scriptfiles) {
+			$filesort += $file.BaseName.Split("_")[2]
+		}
+
+		$currentdate = $filesort | Sort-Object -Descending | Select-Object -First 1
+
+		foreach ($file in $scriptfiles) {
+			if ($file.BaseName.Split("_")[2] -eq $currentdate) {
+				$parsedscript = Invoke-DbaDiagnosticQueryScriptParser -filename $file.fullname -NoQueryTextColumn:$NoQueryTextColumn -NoPlanColumn:$NoPlanColumn -NoColumnParsing:$NoColumnParsing
+
+				$newscript = [pscustomobject]@{
+					Version = $file.Basename.Split("_")[1]
+					Script  = $parsedscript
+				}
+				$scriptversions += $newscript
+			}
+		}
+	}
+
+	process {
+		if (Test-FunctionInterrupt) { return }
+		foreach ($instance in $SqlInstance) {
+			$counter = 0
+			try {
+				Write-Message -Level Verbose -Message "Connecting to $instance"
+				$server = Connect-SqlInstance -SqlInstance $instance -SqlCredential $sqlcredential
+			}
+			catch {
+				Stop-Function -Message "Failure" -Category ConnectionError -ErrorRecord $_ -Target $instance -Continue
+			}
+
+			Write-Message -Level Verbose -Message "Collecting diagnostic query data from server: $instance"
+
+			if ($server.VersionMinor -eq 50) {
+				$version = "2008R2"
+			}
+			else {
+				$version = switch ($server.VersionMajor) {
+					9 { "2005" }
+					10 { "2008" }
+					11 { "2012" }
+					12 { "2014" }
+					13 { "2016" }
+					14 { "2017" }
+				}
+			}
+
+			if (!$instanceOnly) {
+				if (@($DatabaseName).count -eq 0) {
+					$databases = $server.Query("Select Name from sys.databases where name not in ('master', 'model', 'msdb', 'tempdb')")
```

Comment:
CS_AS instances would fail here, use name instead of Name

Reply:
Case sensitivity strikes! Nice catch.